Fixing git ai issues

St4rman opened the pull request:
This PR is for enemies being damageable and some added states.

* The IDamageable is an interface that you can inherit from and that has the latent TakeDamage(int damage) function that both players and enemies can use. EStats is the new enemyStats script and PlayerStats is a debug script i made rn to test player damage to enemies
* Float UI and State UI are for debugs, and This branch also has player being chased 
* Press H to kill an enemy slowly but surely

St4rman committed:
Initial Commit

St4rman committed:
Update .gitignore

St4rman committed:
general push

St4rman committed:
Debug Ui to make states easier

* I shouldnt have spent this much time on this but i did. I am sorry.

St4rman committed:
Chasing

* self explanatory

St4rman committed:
multiple enemies

St4rman committed:
Update E_Wanderer.prefab

St4rman committed:
fixes

St4rman committed:
Update Wanderer.cs

St4rman committed:
Basic Damaging Capabilities

Interfaces

St4rman committed:
fixes

St4rman committed:
Player do damage

St4rman committed:
Instantiating scripts instead of having them pre set

St4rman committed:
Attack bool

St4rman committed:
final stuff

St4rman committed:
Final clean up

okay this is finally clean

St4rman committed:
added stick

DelCherub commented on quirklike/Assets/AI/FSM/States/Wanderer Specific/IdleState.cs:
```diff
@@ -0,0 +1,33 @@
+
+    public void Tick()
+    { 
+        IDLE_TIME --;
```
Won't this be framerate dependent? 1 tick = 1 Update() so this should probably be a float instead and use deltaTime?

St4rman replied:
This is mostly debug ngl, we can use deltaTime in the actual build :] the idle is just there to have the wake up animation tbh with you. But yeah can change it to detla time if really required

DelCherub commented on quirklike/Assets/AI/Standard Scripts/EnemyStats.cs:
```diff
@@ -6,22 +6,28 @@
 //but it is a problem for future me to worry about
 
-public class EnemyStats : MonoBehaviour {
+public class EnemyStats : MonoBehaviour, IDamageable 
```
What is the difference between EStats and EnemyStats? like what purpose do they each have

St4rman replied:
Enemystats is not suppsoed to be there. Estats is the new version. I tried changing Enemystats but decided to write a new script instead 


Estats is basically supposed to be the generalized stats script that you can stick onto any enemy and then interface with the TakeDamage function, and getposition if you need it. If you take a look at playerstats (also an example rn for debug) you can see how it looks for estats and checks if it has an Idamageable interface. Then all you gotta do is call take damage and itll do said damage

DelCherub reviewed (changes requested):
Looks alright but just kinda want some clarity on a few things :)

DelCherub committed:
Enemy Prefab Fixes

DelCherub committed:
Merge branch 'main' into Fixing-git-Ai-issues

DelCherub committed:
Estats Fixes

- Fixed issues that arose from changing EnemyStats and EStats

DelCherub committed:
Altered

- Added Detection Radius object to enemy so that the physics layer can be set separately.
- Fixed but with TestProjectiles related to RigidBody forces by making the rigidbody kinematic (projectiles that have gravity enabled will have to not be kinematic and reset forces upon disabling.)
- Fixed typo in GameRaycastLayers.

DelCherub committed:
Prefabbed projectile pistol
